frontend-service/src/main/java/com/amazonaws/blox/frontend/operations/CreateEnvironment.java

```diff
@@ -41,12 +46,12 @@
   @ApiOperation(value = "Create a new Environment", notes = CREATE_ENVIRONMENT_NOTES)
   public CreateEnvironmentResponse createEnvironment(
       @ApiParam(name = "cluster", value = CLUSTER_PARAM_DESCRIPTION) @PathVariable String cluster,
-      @ApiParam(required = true) @RequestBody CreateEnvironmentRequest request) {
+      @ApiParam(required = true) @RequestBody CreateEnvironmentRequest request)
+      throws InvalidParameterException, ServiceException, ResourceExistsException {
 
-    // TODO: Stub implementation, replace with dataservice call
-    String id = UUID.randomUUID().toString();
-
-    return CreateEnvironmentResponse.builder().environmentRevisionId(id).build();
+    return mapper.fromDataServiceResponse(
+        dataService.createEnvironment(
```

Comment:
don't we need to map the exceptions into frontend exceptions? and log?

Reply:
Spring exception mapping is complicated, and this PR was already pretty big, so I opted to do it as a followup to this PR. There's an open task for it on #280, which we'll probably have to split into a separate issue, since it's a big chunk of work.